crates/core_arch/src/nvptx/mod.rs

```diff
@@ -11,6 +11,8 @@
 //! [llvm_docs]:
 //! https://llvm.org/docs/NVPTXUsage.html
 
+use core::ffi::c_void;
```

Comment:
`core_arch` is compiled inside libcore by using a `#[path = "..."]` to the `core_arch/src/mod.rs` file (the `lib.rs` is only used when building `core_arch` as a standalone library). 

So this needs to use `crate::ffi::c_void` to be able to compile in both Rust 2015 and Rust 2018 modes inside `libcore`.  You might need to bring the `ffi` module to the top module (check out `lib.rs` and `mod.rs`).

Reply:
I've added a commit to address the issue. That's an incredibly clever solution, I must say :slightly_smiling_face:

Reply:
> That's an incredibly clever solution, I must say

@alexcrichton came up with it 🎉